packages/uiweb/src/lib/components/chat/ChatProfile/PendingMembers.tsx

```diff
@@ -0,0 +1,390 @@
+import ArrowIcon from '../../../icons/CaretUp.svg';
+import { Span, Image, Section, Spinner } from '../../reusables';
+import { pCAIP10ToWallet, shortenText } from '../../../helpers';
+import DismissAdmin from '../../../icons/dismissadmin.svg';
+import AddAdmin from '../../../icons/addadmin.svg';
+import Remove from '../../../icons/remove.svg';
+
+
+interface ShadowedProps {
+  setPosition: boolean;
+}
+
+type PendingMembersProps = {
+  pendingMemberPaginationData: MemberPaginationData;
+  setPendingMemberPaginationData: React.Dispatch<
+    React.SetStateAction<MemberPaginationData>
+  >;
+  pendingMembers: ChatMemberProfile[];
+  setShowPendingRequests: React.Dispatch<React.SetStateAction<boolean>>;
+  showPendingRequests: boolean;
+  count: number;
+  theme: IChatTheme;
+};
+type AcceptedMembersProps = {
+  acceptedMemberPaginationData: MemberPaginationData;
+  setAcceptedMemberPaginationData: React.Dispatch<
+    React.SetStateAction<MemberPaginationData>
+  >;
+  acceptedMembers: ChatMemberProfile[];
+  chatId: string;
+};
+
+const UPDATE_KEYS = {
+  REMOVE_MEMBER: 'REMOVE_MEMBER',
+  ADD_MEMBER: 'ADD_MEMBER',
+  REMOVE_ADMIN: 'REMOVE_ADMIN',
+  ADD_ADMIN: 'ADD_ADMIN',
+} as const;
+
+const SUCCESS_MESSAGE = {
+  REMOVE_MEMBER: 'Removed Member successfully',
+  ADD_MEMBER: 'Group Invitation sent',
+  REMOVE_ADMIN: 'Admin removed successfully',
+  ADD_ADMIN: 'Removed added successfully',
+};
+
+export const PendingMembers = ({
+  pendingMembers,
+  setShowPendingRequests,
+  setPendingMemberPaginationData,
+  showPendingRequests,
+  pendingMemberPaginationData,
+  count,
+  theme,
+}: PendingMembersProps) => {
+  const pendingMemberPageRef = useRef<HTMLDivElement>(null);
+
+  const isInViewportPending = useIsInViewport(pendingMemberPageRef, '1px');
+
+  useEffect(() => {
+    if (
+      !isInViewportPending ||
+      pendingMemberPaginationData.loading ||
+      pendingMemberPaginationData.finishedFetching
+    ) {
+      return;
+    }
+
+    const newPage = pendingMemberPaginationData.page + 1;
+    setPendingMemberPaginationData((prev: MemberPaginationData) => ({
+      ...prev,
+      page: newPage,
+    }));
+    // eslint-disable-next-line no-use-before-define
+  }, [isInViewportPending]);
+
+  if (pendingMembers && pendingMembers.length) {
+    return (
+      <PendingRequestWrapper theme={theme}>
+        <PendingSection
+          onClick={() => setShowPendingRequests(!showPendingRequests)}
+        >
+          <Span fontSize="18px" color={theme.textColor?.modalSubHeadingText}>
+            Pending Requests
+          </Span>
+          <Badge>{count}</Badge>
+
+          <ArrowImage
+            src={ArrowIcon}
+            width={'auto'}
+            setPosition={!showPendingRequests}
+            borderRadius="100%"
+          />
+        </PendingSection>
+
+        <ProfileSection
+          flexDirection="column"
+          flex="1"
+          maxHeight="10rem"
+          overflow="hidden scroll"
```

Comment:
by default even if there are just 1 user and there is no need for scroll it's still showing scroll so I think we should just make it auto instead of keeping it scroll by default

Reply:
yes we should make it auto